Move cta hiding to woo landing page

lsl opened the pull request:
#### Changes proposed in this Pull Request

* Move ctaAction ref juggling out of the FixedNavigationHeader and into the parent woo landing page component.

#### Testing instructions

* No visible change
* Test scrolling on the woo landing page still adds the CTA to the header when you scroll the content one out of view
* Test domains and plugins navigation header still works as expected (neither made use of this behavior)

https://user-images.githubusercontent.com/811776/161899461-cc71855d-e585-4732-903a-c2907f65f0f8.mp4

Fixes #62166

lsl committed:
Move cta hiding to woo landing page

Rolilink reviewed (approved):
LGTM

Rolilink commented:
@lsl I added the `useScrollAboveComponent` hook that can be used to implement the sticky behavior in https://github.com/Automattic/wp-calypso/pull/62532.

lsl committed:
Switch to hook from #62532

lsl committed:
Use forwarded ref instead

lsl commented on client/my-sites/woocommerce/landing-page.tsx:
```diff
@@ -153,12 +152,16 @@ const LandingPage: React.FunctionComponent< Props > = ( { siteId } ) => {
-					{ displayData.action }
-				</Button>
+			<FixedNavigationHeader navigationItems={ navigationItems } ref={ headerRef }>
+				{ isAboveElement && (
```
@Rolilink we might want to reverse the naming on `isAboveElement` and make it `isBelowElement` which is how it should read in this usage: "if target is below reference, show the target". Alternatively we could rename it to `isReferenceAboveTarget` so there isn't any room for confusion.

Rolilink replied:
@lsl, I like the idea, feel free to do this one but it will probably affect the marketplace search so make sure to make the change there too.

lsl commented:
> @lsl I added the `useScrollAboveComponent` hook that can be used to implement the sticky behavior in #62532.

Nice! Updated to use, tests ok. @lupus2k around for a quick :heavy_check_mark: by chance?
